Azure-Samples/chat-with-your-data-solution-accelerator
# [Integrated Vectorization] Changes for Admin Configuration page

## Purpose

* Resolves #762 
* Related to #321
* This PR adds Max page length & overlap length config values that can be modified from Admin configuration page.
* Adds a factory method for Processing the ingested file.
* Added & updated unit tests

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
If 'AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION' is true, user should be able to see two input values for Max page length & overlap length
```
<img width="886" alt="image" src="https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/assets/29505310/1121d4c5-0442-4f33-8e0b-f09d226d2fbe">


## What to Check
Verify that the following are valid
* Any updates to the config values should be saved.
